### ISSUE TITLE
macro block fails when the first line is a `if` statement

### ISSUE THREAD

**Issue by maiha** [crystal-lang/crystal#5724]
Macros causes a syntax error when it begins with `if`.
```crystal
{%
  if true
  end
%}
```

```
Syntax error in test.cr:3: expecting token '%}', not 'NEWLINE'

  end
```

It works when some expression exists in first line.
```crystal
{%
  1
  if true
  end
%}
```

crystal-0.24.1

Best regards,

**Comment by dNitza** [comment]
```crystal
{% if true
   end %}
```
and 
```crystal
{%
  if true
   end %}
```

Are fine though

**Comment by codyjb** [comment]
Potential lead in parsing of expression suffix, can replicate with trailing `;` which otherwise parses fine outside the macro

```ruby
{%
  if true
  end; %}
```
Noting potential region of error

https://github.com/crystal-lang/crystal/blob/84288b7af588c7e00dc004ad21f5c9ec5be50f83/src/compiler/crystal/syntax/parser.cr#L289

**Comment by asterite** [comment]
This is working well in `master`.

**Comment by Sija** [comment]
Fixed by #6382
